generator/src/main/java/com/graphicsfuzz/generator/semanticschanging/InterchangeExprMutationFinder.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2019 The GraphicsFuzz Project Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.graphicsfuzz.generator.semanticschanging;
+
+import com.graphicsfuzz.common.ast.TranslationUnit;
+import com.graphicsfuzz.common.ast.expr.Expr;
+import com.graphicsfuzz.common.ast.type.Type;
+import com.graphicsfuzz.common.util.IRandom;
+import com.graphicsfuzz.generator.fuzzer.Fuzzer;
+import com.graphicsfuzz.generator.fuzzer.templates.IExprTemplate;
+import com.graphicsfuzz.generator.fuzzer.templates.ParenExprTemplate;
+import com.graphicsfuzz.generator.mutateapi.Expr2ExprMutation;
+import com.graphicsfuzz.generator.mutateapi.Expr2ExprMutationFinder;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class InterchangeExprMutationFinder extends Expr2ExprMutationFinder {
+
+  private final IRandom generator;
+
+  public InterchangeExprMutationFinder(TranslationUnit tu,
+                                       IRandom generator) {
+    super(tu);
+    this.generator = generator;
+  }
+
+  @Override
+  protected void visitExpr(Expr expr) {
```

Comment:
As discussed, it looks like this is overriding a generic "visitExpr" visitor method and then giving up on this expression and all its children when the type cannot be identified, even though the type of the children could be know.

Reply:
Thanks for pointing this out.  I just filed #264 which explains why visitExpr is ill-named.  I will do a separate PR to rename it, as that has nothing to do with this PR.